app/javascript/components/displays.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { mathMod } from 'ramda';
+import { DisconnectedMessage, LoadingMessage } from '@messages/message';
+import FullPanel from '@components/full-panel';
+import SidePanel from '@components/side-panel';
+import Twitter from '@widgets/twitter';
+import Numbers from '@widgets/numbers';
+import Weather from '@widgets/weather';
+import Traffic from '@widgets/traffic';
+
+const Wrapper = styled.div`
+  outline: none;
+`;
+
+const widgetElements = {
+  Weather,
+  Twitter,
+  Numbers,
+  Traffic,
+};
+
+const prefetchDelay = 1500;
+
+export const WidgetDisplay = ({
+  stateCallback,
```

Comment:
Would passing in a callback Object as a single argument to WidgetDisplay (like const callbacks = { setPrevWidgetId, setNextWidgetId, . . . } ) be an appropriate solution?

Reply:
Maybe? An object of functions my have gotchas with changing by reference and causing re-renders so it may require a memoization or just pass them as direct properties.

Reply:
I'll pass them in as direct properties